### PR TITLE
[ButtonBar] Refactor ButtonBarBuilder for testing

### PR DESCRIPTION
**Pull request by romoore**
Pulling out part of the ButtonBarBuilder logic to start making it more
testable.  Extracting the initial configuration of MDCButton from a
UIBarButtonItem into a class method and adding unit tests.

### CODE REVIEW

File: components/ButtonBar/tests/unit/ButtonBarBuilderTests.m [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  XCTAssertEqual(destinationButton.tag, defaultTag);
+}
+
+- (void)testConfigureButtonFromButtonItemWithTitle {
+  // Given
+  UIBarButtonItem *itemWithTitle = [[UIBarButtonItem alloc] initWithTitle:@"Title"
+                                                                    style:UIBarButtonItemStylePlain
+                                                                   target:nil
+                                                                   action:nil];
+  itemWithTitle.tag = 0x02468ace;
```
Comment (jgunaratne) [inline components/ButtonBar/tests/unit/ButtonBarBuilderTests.m:59]: Is this an arbitrary tag? Perhaps add a comment here.

File: components/ButtonBar/tests/unit/ButtonBarBuilderTests.m [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                 @"The MDC button should have the same tag as the UIBarButtonItem.");
+}
+
+- (void)testConfigureButtonFromButtonItemWithImage {
+  // Given
+  UIBarButtonItem *itemWithImage = [[UIBarButtonItem alloc] initWithImage:[[UIImage alloc] init]
+                                                                    style:UIBarButtonItemStylePlain
+                                                                   target:nil
+                                                                   action:nil];
+  itemWithImage.tag = 0x13579bdf;
```
Comment (jgunaratne) [inline components/ButtonBar/tests/unit/ButtonBarBuilderTests.m:90]: 0x13579bdf?

File: components/ButtonBar/tests/unit/ButtonBarBuilderTests.m [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+- (void)testConfigureButtonFromButtonItemWithTitleAndImage {
+  // Given
+  UIBarButtonItem *itemWithTitleAndImage =
+      [[UIBarButtonItem alloc] initWithImage:[[UIImage alloc] init]
+                                       style:UIBarButtonItemStylePlain
+                                      target:nil
+                                      action:nil];
+  itemWithTitleAndImage.title = @"Title";
+  itemWithTitleAndImage.tag = 0x13579bdf;
```
Comment (jgunaratne) [inline components/ButtonBar/tests/unit/ButtonBarBuilderTests.m:121]: 0x13579bdf?

**Review by jgunaratne — APPROVED** [review]

File: components/ButtonBar/tests/unit/ButtonBarBuilderTests.m (hunk 3 above)
Comment (romoore) [reply components/ButtonBar/tests/unit/ButtonBarBuilderTests.m:121]: Odd numbers from 1-15

File: components/ButtonBar/tests/unit/ButtonBarBuilderTests.m (hunk 1 above)
Comment (romoore) [reply components/ButtonBar/tests/unit/ButtonBarBuilderTests.m:59]: Even numbers from 0-14

**Pull request merged**